code-challenges.js

```diff
@@ -6,70 +6,255 @@
+// // --------------------5) Copy/paste your code from #4. Refactor your code to include non-string edge cases. Inform your user that the variable passed into the vowel removal function is not a string.
+// // Use the test variables provided below. Expected output: "true is not a string" "42 is not a string" "mCdngMstr"
 
 var refactorTester1 = true
 var refactorTester2 = 42
 var refactorTester3 = "IAmACodingMaster"
 
+const noVowels = (string) => {
 
+    if(typeof string !== "string"){
+        return `${string} is not a string`
+    } // checks to see if the input is a string or not
 
+    let spStr = string.split("")
+    // split up all of the strings in the array
+    let result = []
 
+    // initialize an empty array for later use 
+    for(i = 0; i < spStr.length; i++){
 
-// --------------------6) Create a function that takes in an array of strings and returns the string with the most characters.
-// Expected output: "chopsticks" "stemware"
-
-var utensils = ["fork", "knife", "tongs", "chopsticks", "skewer"]
-var vessels = ["cup", "mug", "stemware", "glass", "tumbler", "tea cup"]
+        // start of for loop
+        if(spStr[i] === "a" || spStr[i] === "e" || spStr[i] === "i" || spStr[i] === "o" || spStr[i] === "u"){ // checks to see if the element in the split up string is a vowel
+            continue; // this continue statement allows the program to ignore the vowels and continue iterating through the string, and with the else statment below, only add the consonants.
+        } else {
+            result.push(spStr[i])
+            // adds every consonant into the result array 
+        }
+    }
+    return result.join("")
+    // returns the joined consonants to pop out a string again 
+}
+console.log(noVowels(true))
+console.log(noVowels(42))
+console.log(noVowels("IAmACodingMaster"))
 
 
 
+// // --------------------6) Create a function that takes in an array of strings and returns the string with the most characters.
+// // Expected output: "chopsticks" "stemware"
 
+var utensils = ["fork", "knife", "tongs", "chopsticks", "skewer"]
+var vessels = ["cup", "mug", "stemware", "glass", "tumbler", "tea cup"]
 
-// --------------------7) Create a function that takes in an array of strings and returns an array with the last letter removed from each string.
-// Expected output: ["Luig", "Peac", "Toa", "Yosh"] ["Ink", "Blink", "Pink", "Clyd"]
+// would like to try and get this one to work 
+const alphaString = (array) => { 
+
+    let maxLength = 0;
+    // setting a variable which will store the max length of a given element within an array
+    let lenArr = [];
+    // initializing an empty array for later use 
+    let ind;
+    // initializing a variable that will set the index of the largest string
+    for(i = 0; i < array.length; i++){
+
+        lenArr.push(array[i].length);
+        // adds the length of the current element to the empty array
+        for(j = 0; j < lenArr.length; j++){
```

Comment:
oooh nested for-loops!  Awesome pseudocode! This logic all works great! Nested for-loops are definitely unavoidable in some places but for this problem it looks like it's doing more work than needed because it's looping through `lenArr` multiple times for each value in `array`. This isn't bad but loops can be a slow process and less loops generally makes code more performant. The main question for this is, does `lenArr` need to be checked for the longest word length while it (`lenArr`) is getting built?